rotkehlchen/chain/ethereum/modules/weth/decoder.py

```diff
@@ -0,0 +1,180 @@
+                    event.counterparty = CPT_WETH
+                    event.event_type = HistoryEventType.DEPOSIT
+                    event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                    event.notes = f'Wrap {deposited_amount} {self.eth.symbol} as {self.weth.symbol}'  # noqa: E501
+
+        if is_interacting_with_protocol is True:
+            return None, None
+
+        out_event = HistoryBaseEntry(
+            event_identifier=transaction.tx_hash,
+            sequence_index=self.base_tools.get_next_sequence_counter(),
+            timestamp=ts_sec_to_ms(transaction.timestamp),
+            location=Location.BLOCKCHAIN,
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.RECEIVE_WRAPPED,
+            asset=self.weth,
+            balance=Balance(amount=deposited_amount),
+            location_label=depositor,
+            counterparty=CPT_WETH,
+            notes=f'Receive {deposited_amount} {self.weth.symbol}',
+        )
+        maybe_reshuffle_events(
+            out_event=out_event,
+            in_event=in_event,
+            events_list=decoded_events + [out_event],
+        )
+        return out_event, None
+
+    def _decode_withdrawal_event(
+            self,
+            tx_log: EthereumTxReceiptLog,
+            transaction: EvmTransaction,
+            decoded_events: List[HistoryBaseEntry],
+            all_logs: List[EthereumTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: Optional[List[ActionItem]],  # pylint: disable=unused-argument
+    ) -> Tuple[Optional[HistoryBaseEntry], Optional[ActionItem]]:
+        withdrawer = hex_or_bytes_to_address(tx_log.topics[1])
+        withdrawed_amount_raw = hex_or_bytes_to_int(tx_log.data[:32])
+        withdrawed_amount = asset_normalized_value(amount=withdrawed_amount_raw, asset=self.eth)
+
+        in_event = None
+        is_interacting_with_protocol = False
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.counterparty in (WETH_CONTRACT, withdrawer) and
+                event.balance.amount == withdrawed_amount and
+                event.asset == self.eth
+            ):
+                if event.counterparty == withdrawer:
+                    is_interacting_with_protocol = True
+                else:
+                    in_event = event
+                    event.counterparty = CPT_WETH
+
+        if is_interacting_with_protocol is True:
+            return None, None
+
+        weth_withdrawal_event = HistoryBaseEntry(
+            event_identifier=transaction.tx_hash,
+            timestamp=ts_sec_to_ms(transaction.timestamp),
+            sequence_index=self.base_tools.get_next_sequence_counter(),
+            location=Location.BLOCKCHAIN,
+            event_type=HistoryEventType.WITHDRAWAL,
+            event_subtype=HistoryEventSubType.REMOVE_ASSET,
```

Comment:
Should subtype be [HistoryEventSubType.RETURN_WRAPPED](https://github.com/prettyirrelevant/rotki/blob/544643ec3b4eb4283f4ba34bd6cbad0e3cc7123f/rotkehlchen/accounting/structures/types.py#L62) instead?